test/pylib/util.py

```diff
@@ -3,6 +3,7 @@
 #
```

Comment:
The title of this commit message is very long. It could be something like "test: move `wait_for_first_completed` to `pylib/util.py`".

You should always try to make titles as short as possible. The only information included should be what you do and where without any details. "which returns on completion of any of the supplied coroutines" is a detail. If it was important, you should mention it below the title in a separate paragraph. But in this case it's not important -- the name of the function is intuitive and the diff already shows what it does.

What is important, however, is **why** you do this change. You should add the explanation below the title: you want to reuse this function in a new test added in the following commit and you want to avoid code duplication.

I think this commit is an important case so I'll elaborate. Writing a commit message like "I do this here because I do that in one of the following commits" may feel a bit weird initially, but it's actually a common pattern. You often want to do a big change (like adding a new test), but you need to do some self-contained adjustments first (like changing the testing framework). Then, you do these adjustments in separate commits before the big change to make the PR more readable -- you don't want one huge commit with many different changes and you can't do the adjustments after the big change because it depends on the adjustments. So, it's the only thing you can do. However, note that a PR structured this way can be very confusing for reviewers because they usually review PRs commit by commit. If for some commit there is no information that one of the following commits is relevant, it's pretty bad.

Reply:
Done, added explanation!